0_portfolio_input_check_functions.R

```diff
@@ -1,54 +1,55 @@
+devtools::load_all()
```

Comment:
I still don't understand what this function does. Its help file describes it as "load_all loads a package. It roughly simulates what happens when a package is installed and loaded with library()."

Which package is it loading? What is the purpose of simulating "what happens when a package is installed and loaded" at the beginning of an analysis process? If the/a package needed to be loaded, why not load it with `library()` here?

Reply:
The PACTA_analysis repo contains an R package (see DESCRIPTION). Here, `devtools::load_all()` sources all R files in R/ and all datasets in data/. It is using the package in "developer mode" instead of "user mode" with `library()`; this is a hack (and likely a bad idea) to avoid the more verbose approach:

```r
source("R/some-file.R")
source("R/another-file.R")
load("data/some-data.rda")
load("data/some-more-data.rda")
```

Reply:
Eventually, we may extract the package elsewhere and use it with `library()`. But for now, I find it convenient to have the R/ functions alongside the PACTA_analysis code so git tracks the refactoring moves I do.

Reply:
OK, I think I understand it now. :-)

But this raises the question for me then, why are this file `0_portfolio_input_check_functions.R` and similar files like `0_global_functions.R` not also in the `./R` sub-directory?

Reply:
Great question. 
I move code to R/ as I touch it, and see an opportunity to extract something that stands alone, or need to test something (like `read_file()`) -- using the convenient paring between R/file.R and tests/testthat/test-file.R that  works automatically with testtaht.

Reply:
So then, would you suggest/support the idea, at least in the long run, of moving scripts like `0_global_functions.R` and/or the functions contained in that script to the `R/` sun-directory?

Just to understand if that is the long-term strategy... because for me I think that was part of the confusion, we are dependent here now on some R-package like structure, functions, tools, etc., but then much of this repo doesn't look anything like an R package to me (not even sure if it would successfully build unless all of these extraneous files were added to `.Rbuildignore`), as you say it looks more like a data science pipeline.

Reply:
> would you suggest/support the idea, at least in the long run, of moving scripts like 0_global_functions.R and/or the functions contained in that script to the R/ sun-directory?

Yes, I think that's one good strategy to refactor this code base. To me, it makes sense because I don't want to change any existing behavior unless I can test its impact. And the structure R/ + tests/testthat makes testing functions easy. I wouldn't go crazy moving all functions; instead I would move functions as we need to work on them (as we did in this in this PR).

> we are dependent here now on some R-package like structure, functions, tools, etc.

Yes, but only minimally and for convenience. The scripts don't _need_ to know that R/ has functions that also belong to a package -- i.e. we could remove the call to `devtools::load_all()` and replace it with calls to `source("R/....R")` and `read_file("data/....")`. And even if we do that, the package structure is convenient because I can test all functions with a keyboard shortcut.

> not even sure if it would successfully build unless all of these extraneous files were added to .Rbuildignore)

Yup, you guessed it -- all these extraneous files are in .Rbuildignore. R CMD check passes cleanly, and I plan to add a gh-actions workflow to check on every push.